omnigibson/prims/entity_prim.py

```diff
@@ -955,76 +956,117 @@ def get_relative_linear_velocity(self):
+        """
+        Gets prim's pose with respect to the specified frame.
+
+        Args:
+            frame (Literal): frame to get the pose with respect to. Default to world. parent frame
+            get position relative to the object parent. scene frame get position relative to the scene.
+
+        Returns:
+            2-tuple:
+                - 3-array: (x,y,z) position in the specified frame
+                - 4-array: (x,y,z,w) quaternion orientation in the specified frame
+        """
+
         # If the simulation isn't running, we should read from this prim's XForm (object-level) properties directly
         if og.sim.is_stopped():
-            return XFormPrim.get_position_orientation(self)
+            return XFormPrim.get_position_orientation(self, frame=frame)
         # Delegate to RigidPrim if we are not articulated
         elif self._articulation_view is None:
-            return self.root_link.get_position_orientation()
+            return self.root_link.get_position_orientation(frame=frame)
         # Sim is running and articulation view exists, so use that physx API backend
         else:
-            positions, orientations = self._articulation_view.get_world_poses()
-            return positions[0], orientations[0][[1, 2, 3, 0]]
+            if frame == "world" or frame == "scene":
+                positions, orientations = self._articulation_view.get_world_poses()
+            else:
+                positions, orientations = self._articulation_view.get_local_poses()
 
-    def set_local_pose(self, position=None, orientation=None):
-        # If kinematic only, clear cache for the root link
-        if self.kinematic_only:
-            self.root_link.clear_kinematic_only_cache()
-        # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
-        if og.sim.is_stopped():
-            return XFormPrim.set_local_pose(self, position, orientation)
-        # Delegate to RigidPrim if we are not articulated
-        elif self._articulation_view is None:
-            self.root_link.set_local_pose(position=position, orientation=orientation)
-        # Sim is running and articulation view exists, so use that physx API backend
-        else:
-            if position is not None:
-                position = np.asarray(position)[None, :]
-            if orientation is not None:
-                orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
-            self._articulation_view.set_local_poses(position, orientation)
-            PoseAPI.invalidate()
+        position, orientation = positions[0], orientations[0][[1, 2, 3, 0]]
+
+        # If we are in a scene, compute the scene-local transform
+        if frame == "scene" and self.scene is not None:
+            position, orientation = T.relative_pose_transform(
+                position, orientation, *self.scene.prim.get_position_orientation()
```

Comment:
same, we should check for frame == "scene" and self.scene is None. Maybe right after enforcing frame is one of the three values?